sklearnex/_device_offload.py

```diff
@@ -73,26 +74,42 @@ def dispatch(obj, method_name, branches, *args, **kwargs):
         patching_status.write_log(queue=q)
         return branches[backend](obj, *hostargs, **hostkwargs, queue=q)
     if backend == "sklearn":
-        patching_status.write_log()
-        return branches[backend](obj, *hostargs, **hostkwargs)
+        if "array_api_dispatch" in get_config() and get_config()["array_api_dispatch"]:
+            # TODO:
+            # logs require update for this branch.
+            patching_status.write_log()
+            return branches[backend](obj, *args, **kwargs)
+        else:
+            patching_status.write_log()
+            return branches[backend](obj, *hostargs, **hostkwargs)
     raise RuntimeError(
         f"Undefined backend {backend} in " f"{obj.__class__.__name__}.{method_name}"
     )
 
 
 def wrap_output_data(func):
+    """
+    Handles Array API input. Converts output of decorated function
+    to input Array API format on the same device.
```

Comment:
This may need to be rewritten, I know what you are getting at, but the english has me confused.

Reply:
Sorry this may not be the best description. Could you please suggest better description here?

Reply:
Thank you for the proposal. Updated.